sshd-sftp/src/main/java/org/apache/sshd/server/subsystem/sftp/AbstractSftpSubsystemHelper.java

```diff
@@ -1530,14 +1530,34 @@ protected void doRemoveDirectory(int id, String path, LinkOption... options) thr
     protected void doRemove(int id, Path p) throws IOException {
         SftpEventListener listener = getSftpEventListenerProxy();
         ServerSession session = getServerSession();
-        listener.removing(session, p);
+        boolean isDirectory = Files.isDirectory(p);
```

Comment:
According to the Javadoc:

>> false if the file does not exist, is not a directory, or it cannot be determined if the file is a directory or not.

This means that if file does not exist to begin with we might declare it a"regular" file and signal the incorrect callback.